File: core_challenges/03_code_to_comment/code_to_comment.py

def find_primes(max):
    sieve = [True] * (max + 1)
    sieve[0] = sieve[1] = False
    
    for i in range(2, int(max ** 0.5) + 1):
        if sieve[i]:
            for j in range(i * i, max + 1, i):
                sieve[j] = False
    
    return [num for num, is_prime in enumerate(sieve) if is_prime]

print(find_primes(30))

File: core_challenges/03_code_to_comment/code_to_comment.js

function findPrimes(max) {
    const sieve = new Array(max + 1).fill(true);
    sieve[0] = sieve[1] = false;
    
    for (let i = 2; i <= Math.sqrt(max); i++) {
        if (sieve[i]) {
            for (let j = i * i; j <= max; j += i) {
                sieve[j] = false;
            }
        }
    }
    
    return sieve.reduce((primes, isPrime, num) => {
        if (isPrime) primes.push(num);
        return primes;
    }, []);
}

console.log(findPrimes(30));